cmd/nodeup/main.go

```diff
@@ -64,17 +68,47 @@ func main() {
 	retries := flagRetries
 
 	for {
-		cmd := &nodeup.NodeUpCommand{
-			ConfigLocation: flagConf,
-			Target:         target,
-			CacheDir:       flagCacheDir,
-			FSRoot:         flagRootFS,
-			ModelDir:       models.NewAssetPath("nodeup"),
-		}
-		err := cmd.Run(os.Stdout)
-		if err == nil {
-			fmt.Printf("success")
-			os.Exit(0)
+		var err error
+		if install {
+			var command []string
+			for i := 0; i < len(os.Args); i++ {
+				s := os.Args[i]
+				if s == "-install" || s == "--install" {
+					continue
+				}
+				if i == 0 {
+					// We could also try to evaluate based on cwd
+					s, err = os.Readlink("/proc/self/exe")
+					if err != nil {
+						glog.Fatalf("error reading /proc/self/exe link: %v", err)
```

Comment:
Could we have more context here with the error message?  The error message is that we cannot read `/proc/self/exe`, but why are we trying to read it.